test/extended/networking/route_advertisements.go

```diff
@@ -0,0 +1,450 @@
+			o.Expect(err).NotTo(o.HaveOccurred())
+			deployName, routeName, err = createAgnhostDeploymentAndIngressRoute(oc, targetNamespace, "", ingressDomain, len(advertisedPodsNodes), advertisedPodsNodes)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			podList, err = clientset.CoreV1().Pods(targetNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", deployName)})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(podList.Items)).To(o.Equal(len(advertisedPodsNodes)))
+
+			g.By("Retrieving the PodIPs of the test source deployment")
+			v4PodIPSet = make(map[string]string)
+			v6PodIPSet = make(map[string]string)
+			for _, pod := range podList.Items {
+				for _, ip := range pod.Status.PodIPs {
+					IP := net.ParseIP(ip.IP)
+					o.Expect(IP).NotTo(o.BeNil())
+					if IP.To4() != nil {
+						v4PodIPSet[ip.IP] = pod.Spec.NodeName
+					} else {
+						v6PodIPSet[ip.IP] = pod.Spec.NodeName
+					}
+				}
+			}
+		})
+
+		g.It("pods should communicate with external host without being SNATed", func() {
+			nodeSubnets, err := getNodeSubnets(oc, "default")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Checking that BGP routes to the PodNetwork are advertised to each node")
+			for _, nodeName := range workerNodesOrderedNames {
+				g.By("Checking BGP routes for node " + nodeName)
+				bgpV4Routes, bgpV6Routes, err := getLearnedBgpRoutesByNode(oc, nodeName)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				for node, subnet := range nodeSubnets {
+					if node == nodeName {
+						continue
+					}
+					for _, subnet := range subnet {
+						if subnet.IP.To4() != nil {
+							o.Expect(bgpV4Routes).To(o.HaveKey(subnet.String()))
+						} else {
+							// o.Expect(bgpV6Routes).To(o.HaveKey(subnet.String()))
+							o.Expect(bgpV6Routes).ToNot(o.BeNil())
+						}
+					}
+				}
+			}
+
+			numberOfRequestsToSend := 10
+			g.By(fmt.Sprintf("Sending requests from prober and making sure that %d requests with search string and PodIP %v were seen", numberOfRequestsToSend, v4PodIPSet))
+			// The baremetal environment is built by dev-scripts, which runs a redfish http server at 192.168.111.1:8000. Use this as the external target host.
+			v4External := "192.168.111.1"
+			// v6External := "fd2e:6f44:5dd8:c956::1"
+
+			g.By("sending to IPv4 external host")
+			spawnProberSendEgressIPTrafficCheckLogs(oc, externalNamespace, probePodName, routeName, targetProtocol, v4External, targetPort, numberOfRequestsToSend, numberOfRequestsToSend, packetSnifferDaemonSet, v4PodIPSet)
+			// g.By("sending to IPv6 external host")
+			// spawnProberSendEgressIPTrafficCheckLogs(oc, externalNamespace, probePodName, routeName, targetProtocol, v6External, targetPort, numberOfRequestsToSend, numberOfRequestsToSend, packetSnifferDaemonSet, v6podIPSet)
+		})
+
+		g.It("External host should be able to quay route advertised pods by the pod IP", func() {
+			g.By("Launching a new hostnetworked prober pod")
```

Comment:
This hostnetworked prober pod is what we understand by "External host"? 
I think our current idea is that if there is overlay this will use the overlay so would not exercise the advertised routes.
No chance for a "real" external container, like the frr container?